bcda/auth/client/oktaclient.go

```diff
@@ -31,94 +42,213 @@ func init() {
 	} else {
 		logger.Info("No Okta log location provided; using default stderr")
 	}
+	// there is no possibility of recovery if we put the call to config() here
+	// because init() is called once when main is being started
+}
 
+// separate from init for testing
+func config() error {
+	// required env vars
+	oktaBaseUrl = os.Getenv("OKTA_CLIENT_ORGURL")
+	oktaServerID = os.Getenv("OKTA_OAUTH_SERVER_ID")
 	oktaToken := os.Getenv("OKTA_CLIENT_TOKEN")
-	if oktaToken == "" {
-		logger.Fatal("No Okta token found; please set OKTA_CLIENT_TOKEN")
+
+	// report missing env vars
```

Comment:
Could we tidy up comments a bit? What's happening here, for example, is evident from the code below.